kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -296,224 +303,400 @@ object Channel:
             _capacity: Int,
             access: Access = Access.MultiProducerMultiConsumer
         )(using initFrame: Frame, allow: AllowUnsafe): Unsafe[A] =
+            if _capacity <= 0 then
+                new Unsafe[A]:
+                    val takes    = new MpmcUnboundedXaddArrayQueue[Promise.Unsafe[Closed, A]](8)
+                    val puts     = new MpmcUnboundedXaddArrayQueue[Put[A]](8)
+                    val isClosed = new java.util.concurrent.atomic.AtomicBoolean()
 
-            new Unsafe[A]:
-                val queue = Queue.Unsafe.init[A](_capacity, access)
-                val takes = new MpmcUnboundedXaddArrayQueue[Promise.Unsafe[Closed, A]](8)
-                val puts  = new MpmcUnboundedXaddArrayQueue[Put[A]](8)
+                    private val closedResult        = Result.fail(Closed("Channel", initFrame, "zero-capacity"))
+                    private def result[A](value: A) = if isClosed.get() then closedResult else Result.succeed(value)
 
-                def capacity = _capacity
+                    def capacity = 0
 
-                def size()(using AllowUnsafe) = queue.size()
+                    def size()(using AllowUnsafe) = result(0)
 
-                def offer(value: A)(using AllowUnsafe) =
-                    val result = queue.offer(value)
-                    if result.contains(true) then flush()
-                    result
-                end offer
+                    def offer(value: A)(using AllowUnsafe) = result(false)
```

Comment:
I think this behavior isn't ideal because there can be a pending take that could be satisfied. It seems we could just attempt to get a pending take from the queue?

Reply:
Ok I added this update (along with others -- see resolved conversations). This is now breaking a bunch of tests. Do we want to revert or change tests to expect different behavior for the zero-capacity case?

Reply:
Could you elaborate on the behavior changes that break the tests?

Reply:
Current implementation of `ZeroCapacityChannel#close` always returns an empty backlog, since there's no `queue`, but since it's `offer` behavior is different, the accounting that's used in tests for comparing the backlog to what's been `offer`ed doesn't translate.

For instance, the test "offer and close" `offer`s a bunch of values and expects to get everything that was successfully offered back when calling `close()`. A zero-capacity channel can successfully offer them, but returns none of them on `close`.

Basically any test that asserts on backlogs from `close()` either fail or are flaky. Some `Hub` tests are failing now too.

Reply:
It seems just a test issue? The behavior sounds ok to me

Reply:
Made backlog assertions conditional on channel size.